### DISCUSSION TITLE
Android prevents my app from catching the server data?

### DISCUSSION THREAD

**Discussion by kanisuki** [airsdk/Adobe-Runtime-Support#2918] [Show and tell]
I have two problems.

I made an android app.
The app has the function that plays music in the web server.

The script is below.

import flash.netURLRequest;
import flash.media.Sound;
import flash.media.SoundChannl;

var request:URLRequest = new URLRequest("https://*****.com/music1.mp3"); //This  URL is false.
Var sound:Sound = new Sound();
var sndCh:SoundChannel;

music1Btn.addEventListener(MouseEvent.CLICK,music1BtnClick);
function music1BtnClick(evt){
	sound.load(request);
	sndCh = sound.play(0,100);
}

animate ver 24.0
AIRSDK ver 50.2.2.2

I previewed the app in PC.
It played the music in the web server.
I changed the app into the android apk with AIR.
I installed the android apk into my phone and BlueStacks5 in my PC.
But the android apk did not play the music in the web server.
Is this caused by AIR?

Next question,
I changed the music in the web server, but the music name was same "music1".
Then I previewed the app, but the app played the former music1.
It never plays the changed music. Why?

**Comment by itlancer** [comment]
@kanisuki 
Did you add `<uses-permission android:name="android.permission.INTERNET"/>` permission to your AIR Android application manifest? https://github.com/airsdk/Adobe-Runtime-Support/issues/2485#issuecomment-1443743235
Without such permission your Android application cannot send network requests.

**Reply by kanisuki** [reply]
Thank you. Android needs permission, I see.
